rust/private/rustc.bzl

```diff
@@ -371,13 +374,23 @@ def get_linker_and_args(ctx, attr, crate_type, cc_toolchain, feature_configurati
     """
     user_link_flags = get_cc_user_link_flags(ctx)
 
-    if crate_type == "proc-macro":
+    if crate_type in ("bin") or rustdoc:
+        is_linking_dynamic_library = False
+        action_name = CPP_LINK_EXECUTABLE_ACTION_NAME
+    elif crate_type in ("cdylib", "dylib"):
```

Comment:
`cdylib` actually links in all its dependencies, it should be in the same bucket as `proc-macro`, asking for  `CPP_LINK_DYNAMIC_LIBRARY_ACTION_NAME` flags .